appeals/web/src/server/appeals/appeal-details/appellant-case/outcome-incomplete/outcome-incomplete.controller.js

```diff
@@ -80,13 +82,37 @@ const renderIncompleteReason = async (request, response) => {
  */
 const renderUpdateDueDate = async (request, response) => {
 	const { body, currentAppeal, errors } = request;
-
 	let dueDateDay, dueDateMonth, dueDateYear;
 
 	if (request.session.webAppellantCaseReviewOutcome?.updatedDueDate) {
 		dueDateDay = request.session.webAppellantCaseReviewOutcome.updatedDueDate.day;
 		dueDateMonth = request.session.webAppellantCaseReviewOutcome.updatedDueDate.month;
 		dueDateYear = request.session.webAppellantCaseReviewOutcome.updatedDueDate.year;
+	} else {
+		const appellantCase = await appellantCaseService
+			.getAppellantCaseFromAppealId(
+				request.apiClient,
+				currentAppeal.appealId,
+				currentAppeal.appellantCaseId
+			)
+			.catch((error) => logger.error(error));
+
+		if (appellantCase?.applicationDecisionDate) {
+			const defaultDueDate = calculateIncompleteDueDate(
+				appellantCase.applicationDecisionDate,
+				currentAppeal.appealType
+			);
+
+			if (defaultDueDate && isAfter(defaultDueDate, new Date())) {
```

Comment:
i think we shouldn't use `new Date()` anywhere in the code, besides `dates.js` and need to be more explicit here. what time are we comparing - is it today's date at midnight? if so we should use `getTodaysISOString`. Is it today's date at 23:59 (since we are comparing due dates? then we should use something like in [here](https://github.com/Planning-Inspectorate/appeals-back-office/blob/main/appeals/web/src/server/appeals/appeal-details/appellant-case/appellant-case.mapper.js#L618 )
```
dayMonthYearHourMinuteToISOString({
  ...updatedDueDate,
  hour: DEADLINE_HOUR,
  minute: DEADLINE_MINUTE
}
```

or maybe even better, update `getTodaysISOString` to accomodate deadline times. Something like this:

```
/**
 * returns today's ISO string with today's day, month, year at Europe/london midnight time (00:00)
 * @returns {string}
 */
export function getTodaysISOString(atMidnight = true) {
	const nowISOString = new Date().toISOString();

	const { year, month, day } = dateISOStringToDayMonthYearHourMinute(nowISOString);

	let hour = 0;
	let minute = 0;

	if (!atMidnight) {
		hour = DEADLINE_HOUR;
		minute = DEADLINE_MINUTE;
	}

	return dayMonthYearHourMinuteToISOString({
		day,
		month,
		year,
		hour,
		minute
	});
}
```

Reply:
yep, `getTodaysISOString` should be enough for this